test/e2e/network/netpol/network_policy_api.go

```diff
@@ -209,4 +197,63 @@ var _ = common.SIGDescribe("Netpol API", func() {
 		framework.ExpectNoError(err)
 		framework.ExpectEqual(len(nps.Items), 0, "filtered list should have 0 items")
 	})
+
+	/*
+		Release: v1.21
+		Testname: NetworkPolicy support EndPort Field
+		Description:
+		- EndPort field cannot be defined if the Port field is not defined
+		- EndPort field cannot be defined if the Port field is defined as a named (string) port.
+		- EndPort field must be equal or greater than port.
+	*/
+	ginkgo.It("should support creating NetworkPolicy API with endport field [Feature:NetworkPolicyEndPort]", func() {
+		ns := f.Namespace.Name
+		npClient := f.ClientSet.NetworkingV1().NetworkPolicies(ns)
+
+		ginkgo.By("EndPort field cannot be defined if the Port field is not defined.")
+		var endport int32 = 20000
+		egressRule := networkingv1.NetworkPolicyEgressRule{}
+		egressRule.Ports = append(egressRule.Ports, networkingv1.NetworkPolicyPort{EndPort: &endport})
+		npTemplate := GenNetworkPolicy(SetGenerateName("e2e-example-netpol-endport-validate"),
+			SetObjectMetaLabel(map[string]string{"special-label": f.UniqueName}),
+			SetSpecPodSelectorMatchLabels(map[string]string{"pod-name": "test-pod"}),
+			SetSpecEgressRules(egressRule))
+		_, err := npClient.Create(context.TODO(), npTemplate, metav1.CreateOptions{})
+		framework.ExpectError(err, "request template:%v", npTemplate)
+
+		ginkgo.By("EndPort field cannot be defined if the Port field is defined as a named (string) port.")
+		egressRule = networkingv1.NetworkPolicyEgressRule{}
+		egressRule.Ports = append(egressRule.Ports, networkingv1.NetworkPolicyPort{Port: &intstr.IntOrString{Type: intstr.String, StrVal: "serve-80"}, EndPort: &endport})
+		npTemplate.Spec.Egress = []networkingv1.NetworkPolicyEgressRule{egressRule}
+		_, err = npClient.Create(context.TODO(), npTemplate, metav1.CreateOptions{})
+		framework.ExpectError(err, "request template:%v", npTemplate)
+
+		ginkgo.By("EndPort field must be equal or greater than port.")
+		ginkgo.By("EndPort field is greater than port.")
```

Comment:
in this test endport is 20000 that is lower than port 30000, 
did you mean the opposite in the message?

Reply:
when EndPort field is greater than port,  I expect Error below.

```
framework.ExpectError(err, "request template:%v", npTemplate)
```

Reply:
@aojea you are right.  I will update it.